src/Sarif.UnitTests/Converters/AndroidStudioConverterTests.cs

```diff
@@ -137,238 +139,291 @@ public void AndroidStudioConverter_ConvertToSarifIssue_HintsAreStapledToDescript
             result.Properties.Should().Equal(new Dictionary<string, string> {
                 {"severity", "warning"}
             });
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertToSarifIssue_MultiplePropertiesArePersisted()
+        public void AndroidStudioConverter_ConvertToSarifResult_MultiplePropertiesArePersisted()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.AttributeKey = "key";
             builder.Severity = "warning";
             var uut = new AndroidStudioProblem(builder);
-            Result result = AndroidStudioConverter.ConvertProblemToSarifIssue(uut);
+            Result result = new AndroidStudioConverter().ConvertProblemToSarifResult(uut);
             result.Properties.Should().Equal(new Dictionary<string, string> {
                 {"severity", "warning"},
                 {"attributeKey", "key"}
             });
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertSarifIssue_RecordsTopLevelFileAsSourceFile()
+        public void AndroidStudioConverter_ConvertSarifResult_RecordsTopLevelFileAsSourceFile()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.File = "expected_file.java";
             builder.EntryPointType = "file";
             builder.EntryPointName = "bad_file.java";
-            Location loc = GetLocationForBuilder(builder);
-            Assert.AreEqual("expected_file.java", loc.ResultFile.Uri.ToString());
+            Location loc = GetLocationInfoForBuilder(builder).Location;
+            loc.ResultFile.Uri.ToString().Should().Be("expected_file.java");
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertSarifIssue_DoesNotRecordTopLevelEntryPointAsSourceFile()
+        public void AndroidStudioConverter_ConvertSarifResult_DoesNotRecordTopLevelEntryPointAsSourceFile()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.File = null;
             builder.EntryPointType = "file";
             builder.EntryPointName = "expected_file.java";
-            Location loc = GetLocationForBuilder(builder);
-            Assert.IsNull(loc.ResultFile);
+            Location loc = GetLocationInfoForBuilder(builder).Location;
+            loc.ResultFile.Should().BeNull();
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertSarifIssue_RecordsModuleAsTopLevelIfPresent()
+        public void AndroidStudioConverter_ConvertSarifResult_RecordsModuleAsTopLevelIfPresent()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.File = null;
             builder.Module = "my_fancy_binary";
             builder.EntryPointType = "method";
             builder.EntryPointName = "my_method";
-            Assert.AreEqual(
-                new Location
+
```

Comment:
All the test changes follow this pattern. We need to separately specify a `Location` object and an array of `LogicalLocationComponent`, since the latter is no longer contained in the former.